conversions/src/types.rs

```diff
@@ -0,0 +1,239 @@
+use libsecp256k1::{PublicKeyFormat, SecretKey};
+use num_bigint::BigUint;
+use plonky2::field::{
+    secp256k1_base::Secp256K1Base, secp256k1_scalar::Secp256K1Scalar, types::Field,
+};
+use plonky2_ecdsa::curve::{
+    curve_types::AffinePoint,
+    ecdsa::{ECDSAPublicKey, ECDSASecretKey, ECDSASignature},
+    secp256k1::Secp256K1,
+};
+use rand::rngs::OsRng;
+
+use crate::{error::TypeConversionError, utils::convert_u64_le_to_u8_be};
+
+pub struct PrivateKey(pub SecretKey);
```

Comment:
maybe `pub struct PrivateKey( pub(crate) SecretKey )`? similar for PublicKey and Message

Reply:
Changed, also change the signature of the `Wallet`'s method from 

```rs
fn sign_message(&self, message: &Message) -> Signature;
```

to 

```rs 
fn sign_message(self, message: &Message) -> Signature;
```

following our discussion about reusing a private key for signing. In this way, we are more rigid about the interface.